### ISSUE TITLE
Home Assistant plugin doesn't initialise due to manifest file discrepancy 

### ISSUE THREAD

**Issue by wez** [ninjadev64/OpenDeck#36]
After installing the plugin and re-starting, I see this:

```
[2024-12-14][11:23:21][WARN][opendeck::plugins] Failed to initialise plugin at /home/wez/.config/opendeck/plugins/de.perdoctus.streamdeck.homeassistant.sdPlugin: failed to parse manifest
```

**Comment by wez** [comment]
The manifest contents are:

```json
{
  "Actions": [
    {
      "Icon": "images/actionImage",
      "Name": "Entity (generic)",
      "PropertyInspectorPath": "pi.html",
      "States": [
        {
          "Image": "images/ha-button",
          "TitleAlignment": "bottom",
          "FontSize": 9
        }
      ],
      "SupportedInMultiActions": true,
      "UUID": "de.perdoctus.streamdeck.homeassistant.entity",
      "Controllers": [
        "Keypad",
        "Encoder"
      ],
      "Encoder": {
        "layout": "$A0",
        "StackColor": "#AABBCC"
      }
    },
    {
      "Icon": "images/actionImage",
      "Name": "Entity (custom icons)",
      "PropertyInspectorPath": "pi.html",
      "DisableAutomaticStates": true,
      "States": [
        {
          "Image": "images/ha-button-off",
          "TitleAlignment": "bottom",
          "FontSize": 9
        },
        {
          "Image": "images/ha-button-on",
          "TitleAlignment": "bottom",
          "FontSize": 9
        }
      ],
      "SupportedInMultiActions": true,
      "UUID": "de.perdoctus.streamdeck.homeassistant.dual-state-entity"
    }
  ],
  "UUID": "de.perdoctus.streamdeck.homeassistant",
  "Category": "Home Assistant",
  "CategoryIcon": "images/category",
  "SDKVersion": 2,
  "Author": "Christoph Giesche",
  "CodePath": "plugin.html",
  "Description": "Control your Home Assistant entities with stream deck.",
  "Name": "Home Assistant",
  "Icon": "images/plugin/plugin",
  "URL": "https://github.com/cgiesche/streamdeck-homeassistant",
  "Version": "3.5.4.47",
  "OS": [
    {
      "Platform": "mac",
      "MinimumVersion": "10.11"
    },
    {
      "Platform": "windows",
      "MinimumVersion": "10"
    }
  ],
  "Software": {
    "MinimumVersion": "6.4"
  }
}
```

I've found through trial and error that emptying out the `States` arrays allows the plugin to load, so there is presumably a mismatch in the definition of `ActionState`, but it's not obvious to me what it is

https://github.com/ninjadev64/OpenDeck/blob/e9ea53eb4476c7684cba2e13d5b03dad2406aaee/src-tauri/src/shared.rs#L34-L58

Why doesn't the error message include the underlying serde error message? That would be most helpful in a situation like this!

**Comment by wez** [comment]
Ah, the FontSize is defined as `String` but in this manifest it is numeric

**Comment by ninjadev64** [comment]
Sorry I haven't gotten back to you, it's been a busy week.

If this kind of manifest is accepted by Stream Deck then it should also be by OpenDeck so I'll try and fix it today for 2.3.0.

**Comment by ninjadev64** [comment]
@wez I have fixed this in 9e3d675
